### PR TITLE
Reorder evmc_message fields

### PR DESCRIPTION
**Pull request by chfast**
Resolves https://github.com/ethereum/evmc/issues/123.
Resolves https://github.com/ethereum/evmc/issues/122.

### CODE REVIEW

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -70,58 +70,58 @@ enum evmc_flags
  */
 struct evmc_message
 {
+    /** The kind of the call. For zero-depth calls ::EVMC_CALL SHOULD be used. */
```
Comment (axic) [inline include/evmc/evmc.h]: While this isn't changed in this PR, but I think this comment is invalid, it can be create even for depth 0.

**Comment by axic** [comment]
So didn't #123 say that move `code_hash` from message to `execute`?

**Comment by chfast** [comment]
The `extcodehash` is not useful here because you don't know the address of the code.

**Comment by axic** [comment]
> The extcodehash is not useful here because you don't know the address of the code.

Isn't #123 saying that `execute()` should have a `code_hash` parameter? The code is certainly known there.

**Comment by chfast** [comment]
Yes it says that, but now I prefer to skip it.

**Comment by axic** [comment]
I don't need it myself so OK to remove it.

**Comment by chfast** [comment]
I prefer add features when there is use for them. And execute() together with message is a ton of params anyway.

**Pull request merged**